scripts/deploy.sh

```diff
@@ -192,15 +208,20 @@ EOF
     exit 1
   fi
 
-  src_path=$(realpath "$simulator_path/src/aoai-simulated-api")
-
-  # create a tik_token_cache folder to avoid failure in the build
-  mkdir -p "$src_path/tiktoken_cache"
+  set +e
+  existing_image=$(az acr repository show --name $acr_name --image "aoai-simulated-api" --output json 2>&1)
+  set -e
 
-  az acr login --name "$acr_name"
-  az acr build --image "${acr_login_server}/aoai-simulated-api:latest" --registry "$acr_name" --file "$src_path/Dockerfile" "$src_path"
-  
-  echo -e "\n"
+  if echo "$existing_image" | jq . > /dev/null 2>&1; then
+    if [[ "${FORCE_SIMULATOR_BUILD}" != "true" ]]; then
+      echo "Simulator docker image previously pushed. Skipping build."
+    else
+      echo "Simulator docker image previously pushed. Forcing build."
+      build_image "$simulator_path" "$acr_name" "$acr_login_server"
+    fi
+  else
+    build_image "$simulator_path" "$acr_name" "$acr_login_server"
```

Comment:
It might be nice to add output here to understand the flow?

```suggestion
    echo "No simulator docker image previously pushed. Building."
    build_image "$simulator_path" "$acr_name" "$acr_login_server"
```